.github/workflows/xcframework.yml

```
@@ -0,0 +1,98 @@
+        type: string
+        required: true
+      scheme:
+        description: 'The project scheme that should be used for the archiving process.'
+        type: string
+        required: true
+      version:
+          description: 'The version number of the XCFramework embedded in the XCArchives. This version number is also used for the release tag.'
+          type: string
+          required: true
+      configuration:
+        description: 'The build configuration to use when archiving the scheme, either Debug or Release.'
+        type: string
+        required: false
+        default: 'Release'
+      runsonlabels:
+          description: 'JSON-based collection of labels indicating which type of github runner should be chosen.'
+          required: false
+          type: string
+          default: '["macos-13"]'
+  
+
+jobs:
+  build-xcarchive:
+    name: Build XCArchive
+    uses: ./.github/workflows/archive.yml
+    with:
+      archiveName: ${{ inputs.XCArchiveName }}
+      workspaceFile: ${{ inputs.workspaceFile }}
+      scheme: ${{ inputs.scheme }}
+      version: ${{ inputs.version }}
+      configuration: ${{ inputs.configuration }}
+      runs-on: ${{ fromJson(inputs.runsonlabels) }}
+  create-xcframework-and-release:
+    name: Build XCFramework
+    runs-on: ${{ fromJson(inputs.runsonlabels) }}
+    needs: build-xcarchive
+    steps:
+    - uses: actions/checkout@v3
+    - uses: maxim-lobanov/setup-xcode@v1
+      with:
+        xcode-version: latest-stable
+    - name: Check environment
+      run: |
+          xcodebuild -version
+          swift --version
+          echo Release version: ${{ inputs.version }}
+    - uses: actions/download-artifact@v3
+      with:
+        path: ./.build
+    - name: Create XCFramework
+      run: |
+          rm -rf ${{ inputs.XCFrameworkName }}.xcframework
+          xcodebuild -create-xcframework \
+            -framework ./.build/${{ inputs.XCArchiveName }}-iphoneos.xcarchive/Products/Library/Frameworks/${{ inputs.XCArchiveName }}.framework \
+            -framework ./.build/${{ inputs.XCArchiveName }}-iphonesimulator.xcarchive/Products/Library/Frameworks/${{ inputs.XCArchiveName }}.framework \
+            -output ${{ inputs.XCFrameworkName }}.xcframework
+          rm -rf .build
+    - name: Commit and push XCFramework
+      uses: EndBug/add-and-commit@v9
+      with:
+        add: ${{ inputs.XCFrameworkName }}.xcframework
+        message: Create XCFramework for release ${{ inputs.version }}
+        tag: '${{ inputs.version }} --force'
+        tag_push: '--force'
+    - name: Create Artifacts
+      run: |
+          tar -zcvf ${{ inputs.XCFrameworkName }}.xcframework.tar.gz ${{ inputs.XCFrameworkName }}.xcframework
+    - name: Create Release
+      uses: softprops/action-gh-release@v1
+      with:
+        tag_name: ${{ inputs.version }}
+        generate_release_notes: true
+        fail_on_unmatched_files: true
+        files: ${{ inputs.XCFrameworkName }}.xcframework.tar.gz
```

Would suggest to add a final line break here:
```suggestion
        files: ${{ inputs.XCFrameworkName }}.xcframework.tar.gz

```